aspnetcore/tutorials/first-mvc-app/adding-view.md

```diff
@@ -123,37 +148,65 @@ Examine the *Views/_ViewStart.cshtml* file:
 
-Controllers are responsible for providing the data required in order for a view template to render a response. A best practice: View templates should **not** perform business logic or interact with a database directly. Rather, a view template should work only with the data that's provided to it by the controller. Maintaining this "separation of concerns" helps keep the code clean, testable, and maintainable.
+Controllers are responsible for providing the data required in order for a view template to render a response.
 
-Currently, the `Welcome` method in the `HelloWorldController` class takes a `name` and a `ID` parameter and then outputs the values directly to the browser. Rather than have the controller render this response as a string, change the controller to use a view template instead. The view template generates a dynamic response, which means that appropriate bits of data must be passed from the controller to the view in order to generate the response. Do this by having the controller put the dynamic data (parameters) that the view template needs in a `ViewData` dictionary that the view template can then access.
+View templates should **not**:
 
-In *HelloWorldController.cs*, change the `Welcome` method to add a `Message` and `NumTimes` value to the `ViewData` dictionary. The `ViewData` dictionary is a dynamic object, which means any type can be used; the `ViewData` object has no defined properties until you put something inside it. The [MVC model binding system](xref:mvc/models/model-binding) automatically maps the named parameters (`name` and `numTimes`) from the query string in the address bar to parameters in your method. The complete *HelloWorldController.cs* file looks like this:
+* Do business logic
+* Interact with a database directly.
 
-[!code-csharp[](~/tutorials/first-mvc-app/start-mvc/sample/MvcMovie/Controllers/HelloWorldController.cs?name=snippet_5)]
+A view template should work only with the data that's provided to it by the controller. Maintaining this "separation of concerns" helps keep the code:
+
+* Clean.
+* Testable.
+* Maintainable.
+
+Currently, the `Welcome` method in the `HelloWorldController` class:
+
+* Takes a `name` and a `ID` parameter.
+* Outputs the values directly to the browser.
+
+Rather than have the controller render this response as a string, change the controller to use a view template instead. The view template generates a dynamic response, which means that appropriate bits of data must be passed from the controller to the view to generate the response. Do this by having the controller put the dynamic data (parameters) that the view template needs, in a `ViewData` dictionary. The view template can then access the dynamic data.
```

Comment:
```suggestion
Rather than have the controller render this response as a string, change the controller to use a view template instead. The view template generates a dynamic response, which means that appropriate data must be passed from the controller to the view to generate the response. Do this by having the controller put the dynamic data (parameters) that the view template needs in a `ViewData` dictionary. The view template can then access the dynamic data.
```